streaming/index.js

```diff
@@ -238,15 +257,34 @@ const startServer = async () => {
       // Write the HTTP response manually:
       socket.end(`HTTP/1.1 ${statusCode} ${http.STATUS_CODES[statusCode]}\r\n${Object.keys(headers).map((key) => `${key}: ${headers[key]}`).join('\r\n')}\r\n\r\n`);
 
+      // Finally, log the error:
+      request.log.error({
+        err,
+        res: {
+          statusCode,
+          headers
+        },
+        responseTime: Date.now() - startTime
+      }, err.message);
+
       return;
     }
 
     wss.handleUpgrade(request, socket, head, function done(ws) {
       // Remove the error handler:
       socket.removeListener('error', onSocketError);
 
+      request.log.info("Authenticated request & upgraded to WebSocket connection");
+
+      const wsLogger = logger.child({
```

Comment:
The logger child that we're creating here is explicitly a child of the root logger and not `request.log` as the former would include all the request properties with every websocket log message, which would be excessive. Instead we're just passing through `request.id` for correlation purposes.

I could perhaps rename `websocketLogger` to `websocketRequestLogger` and have a `createWebsocketLogger(request)` here.